specification/trace/semantic_conventions/instrumentation/elasticsearch.md

```diff
@@ -0,0 +1,50 @@
+# Semantic conventions for Elasticsearch
+
+**Status**: [Experimental](../../../document-status.md)
+
+This document defines semantic conventions to apply when creating a span for requests to Elasticsearch.
+
+## Span Name
+
+The **span name** SHOULD be of the format `<endpoint id>`.
+
+The elasticsearch endpoint identifier is used instead of the url path in order to reduce the cardinality of the span 
+name, as the path could contain dynamic values. The endpoint id is the `name` field in the
+[elasticsearch schema](https://raw.githubusercontent.com/elastic/elasticsearch-specification/main/output/schema/schema.json).
+If the endpoint id is not available, the span name SHOULD be `http.request.method`.
+
+## URL path parts
+
+Many Elasticsearch url paths allow dynamic values. These SHOULD be recorded in span attributes in the format
+`db.elasticsearch.path_parts.<key>`, where `<key>` is the url path part name. The implementation SHOULD
+reference the [elasticsearch schema](https://raw.githubusercontent.com/elastic/elasticsearch-specification/main/output/schema/schema.json)
+in order to map the path part values to their names.
+
+| Attribute                           | Type | Description                           | Examples                                                                                 | Requirement Level |
+|-------------------------------------|---|---------------------------------------|------------------------------------------------------------------------------------------|---|
+| `db.elasticsearch.path_parts.<key>` | string | A dynamic value in the url path.      | `db.elasticsearch.path_parts.index=test-index`; `db.elasticsearch.path_parts.doc_id=123` | Conditionally Required: [1] |
+
+**[1]:** when the url has dynamic values
```

Comment:
I assume this is possible to know from the ElasticSearch client libraries?

Reply:
Seems so, from https://github.com/open-telemetry/semantic-conventions/pull/23#discussion_r1221535423. all good.